### ISSUE TITLE
RFC - Implement Failed Job Admin

### ISSUE THREAD

**Issue by josegonzalez** [cakephp/queue#11]
It might be nice to implement a job admin panel that lists all failed jobs. This could be done by having custom middleware/listeners that push the failed jobs into a database table, and then a basic page to view said jobs. We would also need a migration for creating that table, and ideally this would be something a user could generate for use with their app vs coming built-into the queue plugin.

I think @dereuromark has an example [here](https://github.com/dereuromark/cakephp-queue/tree/master/docs#admin-backend), though in his case it also allows viewing more info about all jobs since they are all queued to the database.

In the future, we might even be able to expose other functionality/metrics in such an admin - this might involve work upstream to get the proper interfaces exposed - but I think starting here would be good enough :)

**Comment by ishan-biztech** [comment]
I agree with @josegonzalez here. But instead of just displaying only failed jobs, we should display all the jobs that has been processed, status, etc. with beautiful dashboard. Also in addition we should give users ability to restart any particular queue job that has been failed.

The main thing here is that if we add all of this in this package it will be very tedious to maintain and also not good practise instead I suggest to move this functionality into separate dedicated package which handles all of this task. And this should also be optional like not everyone wants this admin panel to monitor the jobs, workers, etc.

**Comment by dereuromark** [comment]
Refs https://github.com/cakephp/queue/issues/145
